### ISSUE TITLE
missing ops in kernel cause misleading error message

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#4325]
If an op is missing in kernel, error messages printed are not helpful at all.
I have a simple home-made `tf.sequential` model that works with `webgl` and `cpu` backends, but fails with `wasm` backend:

```log
backend.ts:665 Uncaught (in promise) Error: 'step' not yet implemented or not found in the registry. This kernel may not be supported by the tfjs backend you have chosen
    at notYetImplemented (backend.ts:665)
    at BackendWasm.step (backend.ts:418)
    at step.ts:48
    at engine.ts:625
    at engine.ts:433
    at Engine.scopedRun (engine.ts:444)
    at Engine.tidy (engine.ts:431)
    at kernelFunc3 (engine.ts:625)
    at engine.ts:639
    at Engine.scopedRun (engine.ts:444)
```

Finally I've traced it to `recurrentActivation` set to `relu` inside a layer instead of default `hardSigmoid` - you'd never guess that based on the message!

Also, I understand that some ops are not implemented in all kernels, but is there a full matrix of ops per kernel?

Environment: TFJS 2.7.0 in Chrome 87

**Comment by rthadur** [comment]
cc @pyu10055 @lina128

**Comment by vladmandic** [comment]
linking to #4500

**Comment by vladmandic** [comment]
update: tested with **tfjs 2.8.3**, error message has changed but result is still a failure:
```
engine.js:487 Uncaught (in promise) Error: Error running Step: Neither modular kernel nor forward func passed
  runKernelFunc	@	engine.js:487
  runKernel	@	engine.js:391
  step_	@	step.ts:47
  step__op	@	operation.ts:51
  x	@	Relu_grad.ts:29
  (anonymous)	@	tape.js:128
  (anonymous)	@	engine.js:323
  scopedRun	@	engine.js:333
  tidy	@	engine.js:322
  (anonymous)	@	engine.js:843
  backpropagateGradients	@	tape.js:128
  (anonymous)	@	engine.js:841
  (anonymous)	@	engine.js:323
  scopedRun	@	engine.js:333
  tidy	@	engine.js:322
  gradients	@	engine.js:837
  variableGrads	@	gradients.js:252
  computeGradients	@	optimizer.js:82
  minimize	@	optimizer.js:38
  (anonymous)	@	training.js:1118
  (anonymous)	@	training_tensors.js:197
  (anonymous)	@	engine.js:323
  scopedRun	@	engine.js:333
  tidy	@	engine.js:322
  tidy	@	globals.js:175
  fitLoop	@	training_tensors.js:188
```

**Comment by pyu10055** [comment]
@vladmandic Step op was added to WASM in the latest (2.8.3), can you give it another try, see the error message makes more sense now? thanks

**Comment by vladmandic** [comment]
@pyu10055 i've noticed it was added and i've tried, but i can't make sense of the new error message:
```
engine.js:487 Uncaught (in promise) Error: Error running Step: Neither modular kernel nor forward func passed
```

**Comment by lina128** [comment]
Hi @vladmandic, the missing op is implemented and merged into our codebase, but hasn't been released yet. I will be in 2.8.4, likely happen later this week.

**Comment by vladmandic** [comment]
@lina128 thanks - i thought it was released in tfjs 2.8.3, my bad. i'll wait for 2.8.4

**Comment by lina128** [comment]
No problem, @vladmandic. You can track what is released using our release notes: https://github.com/tensorflow/tfjs/releases/tag/tfjs-v2.8.3

**Comment by vladmandic** [comment]
closing as fixed in tfjs 2.8.5
